src/snowcli/output/printing.py

```diff
@@ -67,24 +68,51 @@ def _print_formatted(data: List[Dict], columns: Optional[List[str]] = None):
         raise Exception(f"Unknown {output_format} format option")
 
 
-def print_data(data: List[Dict], columns: Optional[List[str]] = None) -> None:
+def print_data(
+    data: t.Iterable[t.Dict], columns: t.Optional[t.List[str]] = None
+) -> None:
     if columns is not None:
         data = [{k: v for k, v in raw.items() if k in columns} for raw in data]
     _print_formatted(data, columns)
 
 
-def _print_table(data: List[Dict], columns: Optional[List[str]] = None):
+def _print_table(data: t.Iterable[t.Dict], columns: t.Optional[t.List[str]] = None):
     from rich.table import Table
 
     if not data:
         print("No data")
         return
 
-    columns = columns or list(data[0].keys())
-
     table = Table(show_header=True, box=box.ASCII)
-    for column in columns:
-        table.add_column(column)
     for row in data:
+        if not columns:
+            columns = list(row.keys())
```

Comment:
You will assign list of columns for each row, this list be the same for all rows.